components/server/ee/src/billing/billing-service.ts

```diff
@@ -40,50 +44,45 @@ export class BillingService {
-            attributionId: AttributionId.render(attributionId),
-            startedTimeOrder: SortOrder.Descending,
-        });
-        const totalUsage = allSessions.map((s) => s.credits).reduce((a, b) => a + b, 0);
-        if (totalUsage >= costCenter.spendingLimit) {
-            return {
-                reached: true,
-                attributionId,
-            };
-        } else if (totalUsage > costCenter.spendingLimit * 0.8) {
-            return {
-                reached: false,
-                almostReached: true,
-                attributionId,
-            };
+        if (attributionId.kind === "team") {
+            const upcomingInvoice = await this.getUpcomingInvoice(attributionId.teamId);
+            const currentUsage = upcomingInvoice.getCredits();
+            if (currentUsage >= costCenter.spendingLimit) {
+                log.info({ userId: user.id }, "Spending limit reached", {
+                    attributionId,
+                    currentUsage,
+                    spendingLimit: costCenter.spendingLimit,
+                });
+                return {
+                    reached: true,
+                    attributionId,
+                };
+            } else if (currentUsage > costCenter.spendingLimit * 0.8) {
+                log.info({ userId: user.id }, "Spending limit almost reached", {
+                    attributionId,
+                    currentUsage,
+                    spendingLimit: costCenter.spendingLimit,
+                });
+                return {
+                    reached: false,
+                    almostReached: true,
+                    attributionId,
+                };
+            }
         }
+
+        if (attributionId.kind === "user") {
+            // TODO
+        }
+
         return {
             reached: false,
             attributionId,
         };
     }
 
-    // TODO (gpl): Replace this with call to stripeService.getInvoice()
-    async listBilledUsage(req: BillableSessionRequest): Promise<BillableSession[]> {
-        const { attributionId, startedTimeOrder, from, to } = req;
-        let timestampFrom;
-        let timestampTo;
-
-        if (from) {
-            timestampFrom = Timestamp.fromDate(new Date(from));
-        }
-        if (to) {
-            timestampTo = Timestamp.fromDate(new Date(to));
-        }
-        const usageClient = this.usageServiceClientProvider.getDefault();
-        const response = await usageClient.listBilledUsage(
-            {},
-            attributionId,
-            startedTimeOrder as number,
-            timestampFrom,
-            timestampTo,
-        );
-        const sessions = response.getSessionsList().map((s) => UsageService.mapBilledSession(s));
-        return sessions;
+    async getUpcomingInvoice(teamId: string): Promise<GetUpcomingInvoiceResponse> {
+        const response = await this.billingServiceClientProvider.getDefault().getUpcomingInvoice(teamId);
```

Comment:
WDYT of using `attributionId` here, and using `ParseAttributionID` on the side of `BillingService`?

Reply:
:point_up: Thought about the same. :brain: :sparkles: :+1: